backend/src/main/kotlin/blue/mild/covid/vaxx/service/VaccinationSlotService.kt

```diff
@@ -0,0 +1,127 @@
+import blue.mild.covid.vaxx.dto.response.VaccinationSlotDtoOut
+import blue.mild.covid.vaxx.error.entityNotFound
+import mu.KLogging
+import org.jetbrains.exposed.sql.Column
+import org.jetbrains.exposed.sql.Op
+import org.jetbrains.exposed.sql.and
+import java.time.Instant
+
+@Suppress("MagicNumber")
+val MAX_FUTURE = Instant.now().plusMillis(2*365*24*60*60*1000)
+val DEFAULT_STATUS = VaccinationSlotStatus.ONLY_FREE
+
+class VaccinationSlotService(
+    private val locationRepository: LocationRepository,
+    private val vaccinationSlotRepository: VaccinationSlotRepository,
+//    private val patientRepository: PatientRepository
+) {
+    private companion object : KLogging()
+
+    @Suppress("TooGenericExceptionThrown", "ThrowsCount")
+    suspend fun addSlots(createDto: CreateVaccinationSlotsDtoIn, locationId: EntityId? = null): List<EntityId> {
+        if (createDto.to < createDto.from.plusMillis(createDto.durationMillis)) {
+            throw Exception("Specified time range is not valid - ${createDto}")
+        }
+
+        if (createDto.locationId != null && locationId != null) {
+            throw Exception("Location mismatch - ${createDto} vs ${locationId}")
+        }
+        val usedLocationId = (createDto.locationId ?: locationId)!!
+
+        val location = locationRepository.getAndMapLocationsBy { Locations.id eq usedLocationId }
+            .singleOrNull() ?: throw entityNotFound<Locations>(Locations::id, usedLocationId)
+
+        var ts = createDto.from
+        val createdIds = mutableListOf<EntityId>()
+        while (ts.plusMillis(createDto.durationMillis).isBefore(createDto.to.plusMillis(1))) {
+            val to = ts.plusMillis(createDto.durationMillis)
+            for (queue in 0 until createDto.bandwidth) {
+                createdIds.add(
+                    vaccinationSlotRepository.addVaccinationSlot(
+                        locationId = location.id,
+                        patientId = null,
+                        queue = queue,
+                        from = ts,
+                        to = to,
+                    )
+                )
+            }
+            ts = to
+        }
+
+        return createdIds
+    }
+
+    /**
+     * Filters the database with the conjunction (and clause) of the given properties.
+     */
+    suspend fun getSlotsByConjunctionOf(
+        id: EntityId? = null,
+        locationId: EntityId? = null,
+        fromMillis: Long? = null,
+        toMillis: Long? = null,
+        status: VaccinationSlotStatus? = DEFAULT_STATUS,
+    ): List<VaccinationSlotDtoOut> {
+        // To Consider: Maybe when id is specified all the other parameter should be
+        val fromI = Instant.ofEpochMilli(fromMillis ?: Instant.EPOCH.toEpochMilli())
+        // MartinLLama: With Instant.MAX it was failing on Exception occurred in the application: long overflow
+        val toI = Instant.ofEpochMilli(toMillis ?: MAX_FUTURE.toEpochMilli())
```

Comment:
I think we had similar problem in `SystemStatisticsService`. The problem is not only overflow but also a fact that Postgres has different size for Timestamp.